Acceptance test "TestAccSystemGetAWSSNSIAMPolicy_basic" fail on different Snowflake account

**Provider Version**

The provider version you are using: `0.23.2`

**Terraform Version**

The version of Terraform you were using when the bug was encountered: `0.14`

**Describe the bug**

* Acceptance test `TestAccSystemGetAWSSNSIAMPolicy_basic` fails when I run acceptance test on my Snowflake account.

```
=== CONT  TestAccSystemGetAWSSNSIAMPolicy_basic
    system_get_aws_sns_iam_policy_acceptance_test.go:22: Step 1/1 error: Check failed: Check 2/2 error: data.snowflake_system_get_aws_sns_iam_policy.p: Attribute 'aws_sns_topic_policy_json' expected "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"1\",\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"arn:aws:iam::494544507972:user/3gmi-s-ssca3411\"},\"Action\":[\"sns:Subscribe\"],\"Resource\":[\"arn:aws:sns:us-east-1:1234567890123456:mytopic\"]}]}", got "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"1\",\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"arn:aws:iam::532029211228:user/154a-s-sgss5425\"},\"Action\":[\"sns:Subscribe\"],\"Resource\":[\"arn:aws:sns:us-east-1:1234567890123456:mytopic\"]}]}"
--- FAIL: TestAccSystemGetAWSSNSIAMPolicy_basic (18.16s)
FAIL
coverage: 71.4% of statements
FAIL    github.com/chanzuckerberg/terraform-provider-snowflake/pkg/datasources  18.201s
?       github.com/chanzuckerberg/terraform-provider-snowflake/pkg/db   [no test files]
```

* This is because expected IAM User ARN is hard coded in the test code `terraform-provider-snowflake/pkg/datasources/system_get_aws_sns_iam_policy_acceptance_test.go` but actual ARB is generated based on actual Snowflake account.

```
resource.TestCheckResourceAttr("data.snowflake_system_get_aws_sns_iam_policy.p", "aws_sns_topic_policy_json",
    `{"Version":"2012-10-17","Statement":[{"Sid":"1","Effect":"Allow",
    "Principal":{"AWS":"arn:aws:iam::532029211228:user/3gmi-s-ssca3411"},
    "Action":["sns:Subscribe"],"Resource":["arn:aws:sns:us-east-1:1234567890123456:mytopic"]}]}`),
),
```

**Expected behavior**

* All of acceptance test should pass.

**Code samples and commands**

```
make test-acceptance
```

**Additional context**

Add any other context about the problem here.

--- a/go.mod
+++ b/go.mod
@@ -21,6 +21,6 @@ require (
 	github.com/stretchr/testify v1.6.1
 	github.com/zclconf/go-cty v1.7.1 // indirect
 	golang.org/x/crypto v0.0.0-20201208171446-5f87f3452ae9
-	golang.org/x/tools v0.0.0-20201211185031-d93e913c1a58
+	golang.org/x/tools v0.1.0
 	google.golang.org/api v0.34.0 // indirect
 )

--- a/pkg/datasources/system_get_aws_sns_iam_policy_acceptance_test.go
+++ b/pkg/datasources/system_get_aws_sns_iam_policy_acceptance_test.go
@@ -17,8 +17,6 @@ func providers() map[string]*schema.Provider {
 }
 
 func TestAccSystemGetAWSSNSIAMPolicy_basic(t *testing.T) {
-	// r := require.New(t)
-
 	resource.ParallelTest(t, resource.TestCase{
 		Providers: providers(),
 		Steps: []resource.TestStep{
@@ -26,8 +24,7 @@ func TestAccSystemGetAWSSNSIAMPolicy_basic(t *testing.T) {
 				Config: policyConfig(),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("data.snowflake_system_get_aws_sns_iam_policy.p", "aws_sns_topic_arn", "arn:aws:sns:us-east-1:1234567890123456:mytopic"),
-					resource.TestCheckResourceAttr("data.snowflake_system_get_aws_sns_iam_policy.p", "aws_sns_topic_policy_json",
-						`{"Version":"2012-10-17","Statement":[{"Sid":"1","Effect":"Allow","Principal":{"AWS":"arn:aws:iam::494544507972:user/3gmi-s-ssca3411"},"Action":["sns:Subscribe"],"Resource":["arn:aws:sns:us-east-1:1234567890123456:mytopic"]}]}`),
+					resource.TestCheckResourceAttrSet("data.snowflake_system_get_aws_sns_iam_policy.p", "aws_sns_topic_policy_json"),
 				),
 			},
 		},

